File: Android/app/src/main/java/alexa4/friendphoto/controllers/PhotosController.java
package alexa4.friendphoto.controllers;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import alexa4.friendphoto.models.Photo;

/**
 * Controller that contains list of photos of one friend
 */
public class PhotosController {
    private static final String TAG = "PhotosController";
    public final ArrayList<Photo> mPhotos = new ArrayList<>();
    // Owner id needs to check is photos already downloaded
    private int mOwnerId = 0;

    /**
     * Set up photos list from input json string
     *
     * @param jsonString json string from response
     */
    public void setPhotos(String jsonString) {
        Log.d(TAG, jsonString);
        mPhotos.clear();

        try {
            JSONObject json = new JSONObject(jsonString).getJSONObject("response");
            JSONArray array = json.getJSONArray("items");
            for (int i = 0; i < array.length(); i++)
                mPhotos.add(new Photo(array.getJSONObject(i)));

        } catch (JSONException e) {
            e.printStackTrace();
            mPhotos.clear();
        }
    }

    /**
     * Check is otherId and this id is equals
     *
     * @return true if otherId == mOwnerId, else false
     */
    public boolean isIdenticalId(int otherId) {
        return mOwnerId == otherId;
    }

    // Default setter
    public void setOwnerId(int ownerId) {
        mOwnerId = ownerId;
    }

    public void dispose() {
        mPhotos.clear();
    }
}

File: Flutter/friend_photo_flutter/lib/src/controllers/photos_controller.dart
import 'dart:convert';

import 'package:friend_photo_flutter/src/models/photo.dart';

/// Controller that contains list of photos of one friend
class PhotosController {
  static final JsonCodec _codec = JsonCodec();

  final List<Photo> photos = [];
  int _ownerId = 0;

  /// Set up photos list from input [jsonString]
  void setPhotos(String jsonString) {
    photos.clear();
    try {
      Map<String, dynamic> json = _codec.decode(jsonString)['response'];
      List list = json['items'];
      list.forEach((photo) => photos.add(new Photo.fromJson(photo)));
    } catch (e) {
      print(e);
      photos.clear();
    }
  }

  /// Check is otherId and this id is equals
  bool isIdential(int otherId) => _ownerId == otherId;

  /// Default setter for id
  set ownerId(int id) => _ownerId = id;

  /// Free memory
  void dispose() {
    photos.clear();
  }
}
